packages/driver/src/cypress/runner.ts

```diff
@@ -492,7 +492,57 @@ const hasOnly = (suite) => {
   )
 }
 
-const normalizeAll = (suite, initialTests = {}, setTestsById, setTests, getRunnableId, getHookId, getOnlyTestId, getOnlySuiteId, createEmptyOnlyTest) => {
+const pruneSuite = (emptySuite) => {
+  emptySuite.cleanReferences()
+
+  if (emptySuite.parent) {
+    const parentSuites = emptySuite.parent.suites.filter((suite) => suite !== emptySuite)
+
+    emptySuite.parent.suites = parentSuites
+    emptySuite.parent._onlySuites = emptySuite.parent._onlySuites.filter((suite) => parentSuites.includes(suite))
+    emptySuite.parent = null
+  }
+}
+
+const pruneEmptySuites = (rootSuite, incrementFoundTestsBy) => {
+  for (const suite of [...rootSuite.suites]) {
+    pruneEmptySuites(suite, incrementFoundTestsBy)
+  }
+
+  if (!rootSuite.suites.length && !rootSuite.tests.length) {
+    pruneSuite(rootSuite)
+  }
+
+  if (rootSuite.tests.length && Cypress.testFilter) {
+    incrementFoundTestsBy(rootSuite.tests.length)
+
+    const tests: any[] = []
+    const onlyTests: any[] = []
+
+    for (const test of rootSuite.tests) {
+      if (Cypress.testFilter.includes(test.fullTitle())) {
```

Comment:
We update the test title when a suite or test is skipped due to the browser 
```
describe('suite 1', { browser: '!edge' }, () => {})
it('t1', { browser: 'chrome' }, () => {})
```
We update the fullTitle here: https://github.com/cypress-io/cypress/blob/zachw/iatr-debug-failures-filter/packages/driver/src/cypress/mocha.ts#L97

I would expect a browser-skipped test to appear as skipped if its original title matched the test filter. It would indicate to users they need to run tests in a different browser.

Reply:
Didn't know this! I'll fix this and add some tests around this behavior.